sections/Discussion/process-reflections.tex

```diff
@@ -0,0 +1,36 @@
+This led to the introduction of a certain set of questions, defined in \autoref{sec:sprint2-conclusion}, which helped to ensure the developers stayed focused and provided the necessary information to facilitate productivity.
+Along with the questions, the stand-up meetings were changed to be conducted daily, which further facilitated the developers staying focused.
+\\\\
+\subsection{Evaluation of Essence}
+One of the main things that was used from Essence was the roles.
+While Essence includes things like potential and filters for the pre-project, these were not introduced until the pre-project phase of the semester had passed, and was deemed less meaningful to use at these stages.\\
+Additionally, since not all group members took the software innovation course where Essence was presented, it took a lot of work to relay the information since the remaining group members would essentially have to learn the curriculum of another course on top of the ones they were already following.
+\\\\
+Our experience with the Essence roles was that they were quite similar to the roles present in the well-known Scrum approach.
+The challenger, in Essence, has a similar role as a product owner, the anchor as the scrum master, and the responder corresponds to being a member of the development team.
+The primary addition that Essence has to roles is the addition of the child role.
+This role was supposed to be a fluctuating role which allowed group members to come with feedback and encourage divergent thinking.\\
+However, since a lot of the pre-project parts of Essence were skipped, the child role did not make a lot of sense and quickly ended up becoming a joke to suggest silly ideas rather than a useful tool.
+\\\\
+Had we known about all parts of Essence before starting the project, there is a large possibility that it would have been significantly more useful, and that going through the pre-project models would have helped in increasing the quality of the final system.
+This assumption is based on the theory that having a clearer idea about the direction and encouraging more in-depth discussions before making choices would have resulted in different choices along the way.
+\\\\
+One of the primary motivators behind using Essence over Scrum is the focus on innovation, where it is encouraged to shift direction along the way, while still staying within a pre-defined problem domain.
+While we attempted to keep this approach in mind, it was difficult due to the strict time constraint that prohibited us from making major changes in the focus, since we would still have to consider the study regulations and whether we could change focus and still be done in time for every possible pivot.
+An example of where we decided to pivot despite the cost involved was in \autoref{sec:update-network-protocol} where the networking protocol was almost entirely reworked from being purely UDP to being a mix of UDP and TCP.
```

Comment:
```suggestion
An example of where we decided to pivot despite the cost involved was in \autoref{sec:update-network-protocol} where the networking protocol was almost entirely reworked from being purely UDP to being a mix of UDP and TCP.
Another part of the project where the use of Essence shines through is the sprint-based structure, which allowed easily pivoting without requiring extensive rewriting of the report.
This structure also shows how the project has progressed and changed throughout the semester, rather than just presenting the final version.
```